cmake/Hipify.cmake

```diff
@@ -1,11 +1,54 @@
 # cmake file to trigger hipify
 
+set(DICT_FILE ${CMAKE_BINARY_DIR}/hipify_output_dict_dump.txt)
+
 # Get the hipify_cli.py source directory from current directory by going to parent directory
-get_filename_component(HIPIFY_SCRIPTS_DIR ${CMAKE_CURRENT_LIST_DIR} DIRECTORY)
+get_filename_component(HIPIFY_DIR ${CMAKE_CURRENT_LIST_DIR} DIRECTORY)
+
+function(write_file_list FILE_SUFFIX INPUT_LIST)
+  message(STATUS "Writing ${FILE_SUFFIX} into file - file_${FILE_SUFFIX}.txt")
+  set(_FULL_FILE_NAME "${CMAKE_BINARY_DIR}/cuda_to_hip_list_${FILE_SUFFIX}.txt")
+  file(WRITE ${_FULL_FILE_NAME} "")
+  foreach(_SOURCE_FILE ${INPUT_LIST})
+    file(APPEND ${_FULL_FILE_NAME} ${CMAKE_CURRENT_SOURCE_DIR}/${_SOURCE_FILE})
+    file(APPEND ${_FULL_FILE_NAME} "\n")
+  endforeach()
+endfunction()
+
+function(get_file_list FILE_SUFFIX OUTPUT_LIST)
+  set(_FULL_FILE_NAME "${CMAKE_BINARY_DIR}/cuda_to_hip_list_${FILE_SUFFIX}.txt")
+  file(STRINGS ${_FULL_FILE_NAME} _FILE_LIST)
+  set(${OUTPUT_LIST}_HIP ${_FILE_LIST} PARENT_SCOPE)
+endfunction()
+
+function(update_list_with_hip_files FILE_SUFFIX DICT_FILE)
+  set(_SCRIPTS_DIR ${HIPIFY_DIR}/tools)
+  set(_FULL_FILE_NAME "${CMAKE_BINARY_DIR}/cuda_to_hip_list_${FILE_SUFFIX}.txt")
+  set(_EXE_COMMAND
+    ${_SCRIPTS_DIR}/replace_cuda_with_hip_files.py
+    --io-file ${_FULL_FILE_NAME}
+    #    --dump-dict-file ${CMAKE_BINARY_DIR}/hipify_output_dict_dump.txt)
+    --dump-dict-file ${DICT_FILE})
+  execute_process(
+    COMMAND ${_EXE_COMMAND}
+    RESULT_VARIABLE _return_value)
+  if (NOT _return_value EQUAL 0)
+    message(FATAL_ERROR "Failed to get the list of hipified files!")
+  endif()
+endfunction()
+
+function(get_hipified_list INPUT_LIST OUTPUT_LIST FILE_SUFFIX DICT_FILE)
+  write_file_list("${FILE_SUFFIX}" "${INPUT_LIST}")
+  update_list_with_hip_files("${FILE_SUFFIX}" ${DICT_FILE})
+  get_file_list("${FILE_SUFFIX}" __temp_srcs)
+  set(${OUTPUT_LIST} ${__temp_srcs_HIP} PARENT_SCOPE)
+endfunction()
+
 set(HIPIFY_COMMAND
-  ${HIPIFY_SCRIPTS_DIR}/hipify_cli.py
+  ${HIPIFY_DIR}/hipify_cli.py
   --project-directory ${PROJECT_SOURCE_DIR}
   --output-directory ${PROJECT_SOURCE_DIR}
+  --dump-dict-file ${DICT_FILE}
```

Comment:
DICT_FILE variable is used as a parameter for the function get_hipified_list(), so I think it is better to keep this variable.
For the parameter value --dump-dict-file, want to keep it as DICT_FILE so that it will be configurable by changing the variable.

Reply:
On further discussion, we decided to remove the DICT_FILE argument from the utility functions since it's available as a CMake variable in this file.

Reply:
Updated as discussed.